uPortal-webapp/src/main/webapp/scripts/portal-analytics.js

```diff
@@ -16,10 +16,119 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-'use strict';
+"use strict";
 
-// Begin Portal Analytics
-window.onload = () => {
-  console.log("Window is loaded in portal analytics from scripts!!!");
-};
-// End Portal Analytics
+/**
+  * Function that captures a click on an outbound link in Analytics.
+  */
+  const outboundClick = (event) => {
+    if((event === undefined) || (event === null)) {
+      //console.log("Tried to process an outbound click, but there was no originating event");
+      return;
+    }
+
+    // Both path and composedPath need to be checked due to browser support
+    const anchorForEvent = (event.path || (event.composedPath && event.composedPath()))[0].closest('a');
+    if((anchorForEvent === undefined) || (anchorForEvent === null)) {
+      //console.log("Tried to process an outbound click, but there was no originating event anchor");
+      return;
+    }
+
+    if ((anchorForEvent.href === undefined) ||
+        (
+          anchorForEvent.href.startsWith('javascript')
+        )) {
+      //console.log("Not firing an analytic event due to href condition: " + eventDetails.href);
+      //console.log(anchorForEvent);
+      return;
+    }
+
+    const eventDetails = {
+      type: 'link',
+      href: anchorForEvent.href,
+      target: anchorForEvent.target,
+    }
+
+    console.log("MILESTONE - would be capturing outbound link for: ");
+    // mem leak - do not merge this!
+    console.log(anchorForEvent);
+    console.log(eventDetails);
+    console.log(event);
+    console.log(event.target);
+
+// TODO - this will likely need to be rejigged. GA must still work when configured.
+//    ga('send', 'event', 'outbound', 'click', title, {
+//      'transport': 'beacon',
+//      'hitCallback': () => {}
+//    })
+  }
+
+  const addPageLevelListeners = () => {
```

Comment:
If these are only intended to be used within this script, it could be good to wrap this in an IIFE to avoid adding these to global scope.

```js
(function () {
   // any functions/vars go here
})();

Reply:
fixed.